lib/stm32/common/usart_common_all.c

```diff
@@ -48,9 +48,18 @@ usart_reg_base
 
 void usart_set_baudrate(uint32_t usart, uint32_t baud)
 {
+
+#if defined STM32F4
+	uint32_t clock = rcc_clock_scale_used->apb1_frequency;
+
+	if ((usart == USART1) ||
+	    (usart == USART6)) {
+		clock = rcc_clock_scale_used->apb2_frequency;
+	}
+#else
```

Comment:
I have suggested in (l0/f0 USART PR #390 ) that we had a chips specific `usart32_t usart_get_clock(usart)` which would be called by the common code. (you can see how it works in the PR) which avoids this sort of ifdef in the code and is a bit cleaner. In the L0/F0 case they can pick from the AHB clock, the LSE clock, or the HSI clock for their baudclock source.